ReactSkia/core_modules/RSkSpatialNavigator.cpp

```diff
@@ -0,0 +1,380 @@
+/*
+* Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+*
+* Use of this source code is governed by a BSD-style license that can be
+* found in the LICENSE file.
+*/
+#include <string>
+#include <algorithm>
+
+#include "RSkSpatialNavigator.h"
+#include "ReactSkia/components/RSkComponent.h"
+#include "ReactSkia/sdk/RNSKeyCodeMapping.h"
+
+namespace SpatialNavigator {
+
+RSkSpatialNavigator* RSkSpatialNavigator::sharedSpatialNavigator_{nullptr};
+std::mutex RSkSpatialNavigator::mutex_;
+
+RSkSpatialNavigator::RSkSpatialNavigator() {
+    std::function<void(rnsKey, rnsKeyAction)> handler = std::bind(&RSkSpatialNavigator::handleKeyEvent, this,
+                                                                std::placeholders::_1, // rnsKey
+                                                                std::placeholders::_2);  // rnsKeyAction
+    eventId_ = NotificationCenter::defaultCenter().addListener("onHWKeyEvent", handler);
+}
+
+RSkSpatialNavigator::~RSkSpatialNavigator() {
+    if(this == sharedSpatialNavigator_)
```

Comment:
should the null assignment be also in lock ?

Reply:
Done.